### ISSUE TITLE
mypy 1.15.0 reports errors with ParamSpec that weren`t reported in 1.14.1

### ISSUE THREAD

**Issue by s17b2-voroneckij** [python/mypy#19090]
**Bug Report**

Consider the following code:
(https://mypy-play.net/?mypy=1.15.0&python=3.12&gist=da93d81e65326f657a96f9174d4fcbee)
```python
from typing import Any

def nop[**T](*args: T.args, **kwargs: T.kwargs) -> dict[str, Any]:
    return {}
```

mypy 1.15.0 reports an error:

main.py:3: error: ParamSpec "T" is unbound  [valid-type]
Found 1 error in 1 file (checked 1 source file)
**To Reproduce**

```python
from typing import Any

def nop[**T](*args: T.args, **kwargs: T.kwargs) -> dict[str, Any]:
    return {}
```

**Expected Behavior**

No error reported

**Actual Behavior**

error reported: 
main.py:3: error: ParamSpec "T" is unbound  [valid-type]
Found 1 error in 1 file (checked 1 source file)

**Your Environment**

- Mypy version used:
- Mypy command-line flags:
- Mypy configuration options from `mypy.ini` (and other config files):
- Python version used:

**Comment by s17b2-voroneckij** [comment]
"Parameter specification variables exist primarily for the benefit of static type checkers. They are used to forward the parameter types of one callable to another callable – a pattern commonly found in higher order functions and decorators. They are only valid when used in Concatenate, or as the first argument to Callable, or as parameters for user-defined Generics. See [Generic](https://docs.python.org/3/library/typing.html#typing.Generic) for more information on generic types." 

Apparently, this is reported correctly
